### PR TITLE
Update chart name and remove legacy Chart packages

### LINKED ISSUE #69: Clean up legacy binaries
We have moved to using the chart-releaser CI job to automatically create releases for new versions of the `op-scim` Chart. Since then the path to obtain the has moved from `https://raw.githubusercontent.com/1password/op-scim-helm/main` to `https://1password.github.io/op-scim-helm` and the binaries are attached to the [releases](https://github.com/1Password/op-scim-helm/releases).

We still have the legacy binaries (ex. `op-scim-2.4.1.tgz`) in the root of the project. These should be cleaned up once they are no longer used and folks have had a chance to update their Helm repo settings.

A good point to do that would be the next minor release of the SCIM bridge, i.e. v2.5.0 (or later).

We should also remove the legacy `index.yaml` from the root of this project.

### LINKED ISSUE #79: Update chart name to op-scim-bridge
Currently the `name` field in the `Chart.yaml` is set to "op-scim". This does not match what the chart should be called, which is "op-scim-bridge".

Further to that we also have several instances in the values and templates where we append "-bridge" to the chart name, for example (from `values.yaml`):
```yaml
scim:
  # name sets the name used for the various constructs used by the SCIM bridge setup
  name: "{{ .Chart.Name }}-bridge"
```

We should fix this inconsistency and set the chart name correctly. We should also ensure that the downstream apps that depend on this chart are updated.

### PR DESCRIPTION
**Pull request by devillexio**
In this PR we update the Chart name from `op-scim` to `op-scim-helm`. We also make the necessary changes to `values.yaml` so that the actual deployed artifacts are still called `op-scim-helm-*`, i.e. same as before.

We remove the legacy Chart packages located in the root of the project. These no longer work with the changes made to the `index.yaml` at the root of the project. The `index.yaml` is kept around for now to provide additional guidance to folks looking to upgrade to the new repo URL.

### Steps to test

1. Install the previous version of the Chart (i.e. v2.4.2) from the repo:
    ```
    helm install op-scim-bridge 1password/op-scim --create-namespace --namespace op-scim-bridge --set scim.credentialsVolume.storageClass=standard
    ```
    - Remember to select the correct storage class for your destination platform, i.e. `standard` for GCP and `do-block-storage` for DO
2. Ensure that the applications are deployed successfully and that there are no errors in the logs:
    - `kubectl get pods --namespace=op-scim-bridge -o wide`
    - `kubectl logs <op-scim-bridge-pod> --namespace=op-scim-bridge`
3. Package the latest chart changes: `helm package charts/op-scim-bridge`
4. Update using the latest chart changes:
    ```
    helm upgrade op-scim-bridge op-scim-bridge-2.5.0.tgz --namespace op-scim-bridge --set scim.credentialsVolume.storageClass=standard
    ```
5. Once again ensure that the applications are deployed successfully and that there are no errors in the logs:
    - `kubectl get pods --namespace=op-scim-bridge -o wide`
    - `kubectl logs <op-scim-bridge-pod> --namespace=op-scim-bridge`
6. [optional] Try the latest chart changes as a fresh install:
    1. Uninstall the previous configuration `helm uninstall op-scim-bridge --namespace=op-scim-bridge`
    2. Ensure no application pods are deployed or running: `kubectl get pods --namespace=op-scim-bridge -o wide`
    3. Install (not upgrade) the latest chart changes:
        ```
        helm install op-scim-bridge op-scim-bridge-2.5.0.tgz --create-namespace --namespace op-scim-bridge --set scim.credentialsVolume.storageClass=standard
        ```
    4. Once again confirm that there are no errors in the deployments or application logs

Resolves #79.
Resolves #69.

### CODE REVIEW

**Comment by devillexio** [comment]
The expectation after this is merged is that for releases <2.5.0 you will install them as `helm install my-release 1password/op-scim ...` and for releases >=2.5.0: `helm install my-release 1password/op-scim-bridge ...`. Notice the addition of the `-bridge` in the latter example.

This will mean that there will be entries for both `op-scim` and `op-scim-bridge` in the `index.yaml` maintained by the chart-releaser for this repo (under the `gh-pages` branch).

Please also help check that the URLs in the README.md and index.yaml point to the correct spots. 🙏

**Review by ag-adampike — APPROVED** [review]
Tested by upgrading the packed version from this PR and by installing fresh (both on DOKS). Worked flawlessly in both cases.


Just as an aside, you can abstract the logs command up to the Deployment object so you don't have to get the Pod name first, i.e.:

```
kubectl logs deploy/op-scim-bridge n=op-scim-bridge
```

File: charts/op-scim-bridge/Chart.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v2
-name: op-scim
-version: 2.4.2
+name: op-scim-bridge
+version: 2.5.0
```
Comment (alicethorne-ab) [inline charts/op-scim-bridge/Chart.yaml:3]: This version also needs to be set in `appVersion`.

Comment (devillexio) [reply charts/op-scim-bridge/Chart.yaml:3]: The `version` and `appVersion` are distinct and don't need to be linked like they were before. We use the `appVersion` to set the `scim.version` in `values.yaml`:
```yaml
scim:
  # ...
  # version of the SCIM bridge
  version: "{{ .Chart.AppVersion }}"
```

And then in turn we use the `scim.version` to set the image tag version for the Docker image in the `deployment.yaml`:
```yaml
      containers:
        - name: {{ tpl .Values.scim.name . }}
          image: {{ .Values.scim.imageRepository }}:{{ tpl .Values.scim.version . }}
```

We don't yet have a version `2.5.0` tagged on the Docker image.

With the latest release of this chart (2.4.2), and going forward, we no longer need to keep the chart version and SCIM bridge application (tag) version in lockstep with each other.

**Comment by devillexio** [comment]
I created #84 to address the bug @alicethorne-ab reported. I will fix that and push up the change shortly.

**Review by alicethorne-ab — APPROVED** [review]
Bug fixed! LGTM. ✅ 

New output correctly has the variable:

```yaml
...
          env:
           ...
            - name: "OP_WORKSPACE_CREDENTIALS"
              value: "/home/opuser/.op/workspace-credentials.json"
          ...
...
```

**Pull request merged**